Blogpost solidity-0.8.20-release-announcement (#215)
ethereum/solidity-blog: _posts/2023-05-10-solidity-0.8.20-release-announcement.md

---
layout: post
published: true
title: 'Solidity 0.8.20 Release Announcement'
date: '2023-05-10'
author: Solidity Team
category: Releases
---

We are excited to announce the latest release of the Solidity Compiler, [Solidity v0.8.20](https://github.com/ethereum/solidity/releases/tag/v0.8.20). 

> This latest version includes a range of improvements and it also introduces the support for Shangaidefining *operators* on [user-defined value types](https://docs.soliditylang.org/en/latest/types.html#user-defined-value-types) (UDVTs)!

Reviewer: Read it again :-).

---
layout: post
published: true
title: 'Solidity 0.8.20 Release Announcement'
date: '2023-05-10'
author: Solidity Team
category: Releases
---

We are excited to announce the latest release of the Solidity Compiler, [Solidity v0.8.20](https://github.com/ethereum/solidity/releases/tag/v0.8.20). 

> This latest version includes a range of improvements in the via-IR pipeline and improves the list of events exposed in the contract ABI, and, most importantly, support for Shanghai!

Author: ```suggestion
This latest version includes a range of improvements in the via-IR pipeline and improves the list of events exposed in the contract ABI, and, most importantly, support for the [Shanghai hard fork](https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/shanghai.md) !
```

For people who might now know immediately what Shanghai is.

Author: @ekpyron are you dismissing the "hard fork" bit, the link or both?

"support for Shanghai" sounds incomplete.

 * Parser: Unary plus is no longer recognized as a unary operator in the AST and triggers an error at the parsing stage (rather than later during the analysis).
 * SMTChecker: Group all messages about unsupported language features in a single warning. The CLI option ``--model-checker-show-unsupported`` and the JSON option ``settings.modelChecker.showUnsupported`` can be enabled to show the full list.
 * SMTChecker: Properties that are proved safe are now reported explicitly at the end of analysis. By default, only the number of safe properties is shown. The CLI option ``--model-checker-show-proved-safe`` and the JSON option ``settings.modelChecker.showProvedSafe`` can be enabled to show the full list of safe properties.
 * Standard JSON Interface: Add experimental support for importing ASTs via Standard JSON.
 * Yul EVM Code Transform: If available, use ``push0`` instead of ``codesize`` to produce an arbitrary value on stack in order to create equal stack heights between branches.

> 

Author: ```suggestion
```

---
layout: post
published: true
title: 'Solidity 0.8.20 Release Announcement'
date: '2023-05-10'
author: Solidity Team
category: Releases
---

We are excited to announce the latest release of the Solidity Compiler, [Solidity v0.8.20](https://github.com/ethereum/solidity/releases/tag/v0.8.20). 
This latest version includes a range of improvements in the via-IR pipeline and improves the list of events exposed in the contract ABI, and, most importantly, support for Shanghai!

### Important Note

This compiler switches the **default** target EVM version to Shanghai, which means that the generated bytecode will include ``PUSH0`` opcodes.
Be sure to [select the appropriate EVM](https://docs.soliditylang.org/en/latest/using-the-compiler.html#setting-the-evm-version-to-target) version in case you intend to deploy on a chain other than mainnet like L2 chains that may not yet support ``PUSH0``,
otherwise, deployment of your contracts will fail.

> 

Author: ```suggestion
```

---
layout: post
published: true
title: 'Solidity 0.8.20 Release Announcement'
date: '2023-05-10'
author: Solidity Team
category: Releases
---

We are excited to announce the latest release of the Solidity Compiler, [Solidity v0.8.20](https://github.com/ethereum/solidity/releases/tag/v0.8.20). 

> This latest version includes a range of improvements and, most importantly, support for Shanghai!
> It also contains performance improvements in the via-IR pipeline and improves the list of events exposed in the contract ABI.

Author: ```suggestion
This latest version includes a range of improvements in the via-IR pipeline and improves the list of events exposed in the contract ABI, and, most importantly, support for Shanghai!
```

### Important Note

> This compiler switches the **default** target EVM version to Shanghai, which means that the generated bytecode will include ``PUSH0`` opcodes.
> Be sure to select the appropriate EVM version in case you intend to deploy on a chain other than mainnet like L2 chains that may not yet support ``PUSH0``,
> otherwise deployment of your contracts will fail.

Author: ```suggestion
This compiler switches the **default** target EVM version to Shanghai, which means that the generated bytecode will include ``PUSH0`` opcodes.
Be sure to [select the appropriate EVM](https://docs.soliditylang.org/en/latest/using-the-compiler.html#setting-the-evm-version-to-target) version in case you intend to deploy on a chain other than mainnet like L2 chains that may not yet support ``PUSH0``,
otherwise, deployment of your contracts will fail.
```

Added a link @ekpyron.

### Experimental AST import via Standard JSON

> An interesting new feature for tooling developers is the ability to import JSON ASTs via the Standard JSON interface using ``SolidityAST`` as input language.
> We release this as an experimental feature so far, mainly due to the lack of proper error reporting on malformed ASTs and the fact that for now we
> generally don't guarantee sound behaviour on invalid input ASTs.
> 
> So while we strongly advice against using the AST import in any production pipeline, and especially against relying on the bytecode generated from imported
> ASTs for production use, we hope that this new feature will enable the easier development of tools in the ecosystem and we encourage giving it a try and
> getting back to us with any feedback.
> 
> Also note that the compiler will perform type-checking and analysis on the imported AST again - to make sure that the compiler honors the transformation you
> may perform on the AST, you can re-export the imported AST and check the result.

Author: ```suggestion
An interesting new feature for tooling developers is the ability to import JSON ASTs via the Standard JSON interface using ``SolidityAST`` as input language.
We have released this as an experimental feature so far, mainly due to the lack of proper error reporting on malformed ASTs and the fact that for now we
generally don't guarantee sound behavior on invalid input ASTs.

So while we strongly advise against using the AST import in any production pipeline, and especially against relying on the bytecode generated from imported
ASTs for production use, we hope that this new feature will enable the easier development of tools in the ecosystem, and we encourage giving it a try and
getting back to us with any feedback.

Also, note that the compiler will perform type-checking and analysis on the imported AST again - to make sure that the compiler honors the transformation you
may perform on the AST, you can re-export the imported AST and check the result.
```

 * AST: Add the ``usedEvents`` field to ``ContractDefinition`` which contains the AST IDs of all events emitted by the contract as well as all events defined and inherited by the contract.

As always, a big thank you to all contributors who helped make this release possible!

## How to install/upgrade
To upgrade to the latest version of the Solidity Compiler, 
simply follow the [installation instructions](https://docs.soliditylang.org/en/latest/installing-solidity.html) available in our documentation. 
Our team has made sure to provide detailed and straightforward steps to make the upgrade process as seamless as possible. 
If you have any questions or run into any issues during the upgrade process, 
don't hesitate to reach out to our [community Matrix channel](https://matrix.to/#/#ethereum_solidity:gitter.im). 

You can download the new version of Solidity here: [v0.8.20](https://github.com/ethereum/solidity/releases/tag/v0.8.20).
If you want to build from the source code, do not use the source archives generated automatically by GitHub, 
instead please use [`solidity_0.8.20.tar.gz`](https://github.com/ethereum/solidity/releases/download/v0.8.20/solidity_0.8.20.tar.gz)

> and take a look at [our documentation on how to build from source](https://docs.soliditylang.org/en/v0.8.20/installing-solidity.html#building-from-source)

Reviewer: Not sure what to do about this now, since the link will be broken :-D. Maybe best to use
```suggestion
and take a look at [our documentation on how to build from source](https://docs.soliditylang.org/en/latest/installing-solidity.html#building-from-source)
```
